src/autopas/selectors/tuningStrategy/FullSearchMPI.h

```diff
@@ -0,0 +1,318 @@
+/**
+ * @file FullSearchMPI.h
+ * @author W. Thieme
+ * @date 17.04.2020
+ */
+
+#pragma once
+
+#include <set>
+#include <sstream>
+
+#include "TuningStrategyInterface.h"
+#include "autopas/containers/CompatibleTraversals.h"
+#include "autopas/selectors/OptimumSelector.h"
+#include "autopas/utils/AutoPasConfigurationCommunicator.h"
+#include "autopas/utils/ExceptionHandler.h"
+
+namespace autopas {
+
+/**
+ * Exhaustive full search of the search space by testing every applicable configuration and then selecting the optimum.
+ * The search space is divided among the ranks.
+ * Use for homogeneous domains.
+ */
+class FullSearchMPI : public TuningStrategyInterface {
+ public:
+  /**
+   * Constructor for the FullSearch that generates the rank's portion of the search space from the allowed options.
+   * @param allowedContainerOptions
+   * @param allowedTraversalOptions
+   * @param allowedDataLayoutOptions
+   * @param allowedNewton3Options
+   * @param allowedCellSizeFactors
+   * @param comm: provide default value mainly for testing, otherwise TuningStrategyFactory handles this
+   */
+  FullSearchMPI(const std::set<ContainerOption> &allowedContainerOptions,
+                const std::set<double> &allowedCellSizeFactors,
+                const std::set<TraversalOption> &allowedTraversalOptions,
+                const std::set<DataLayoutOption> &allowedDataLayoutOptions,
+                const std::set<Newton3Option> &allowedNewton3Options, AutoPas_MPI_Comm comm = AUTOPAS_MPI_COMM_WORLD)
+      : _containerOptions(allowedContainerOptions),
+        _optimalConfig(Configuration()),
+        _configurationCommunicator(AutoPasConfigurationCommunicator()),
+        _localOptimalTime(0),
+        _request(AUTOPAS_MPI_REQUEST_NULL),
+        _allLocalConfigsTested(false),
+        _allGlobalConfigsTested(false),
+        _autopasMPICommunicator(comm) {
+    // @todo distribute the division process, so that not every rank has to traverse all configs
+    // @todo consider using MPI_Scatterv for dividing the search space
```

Comment:
are these todos still valid?

Reply:
> are these todos still valid?

The idea is still relevant, but considering that FullSearchMPI will be deleted soon, I suppose they aren't. I removed them.